File: backend/src/models/DDOTKHUYENMAI.ts
import { Model, DataTypes } from 'sequelize';
import db from '../config/database';

class DDOTKHUYENMAI extends Model {
    public id!: any;
    public name!: any;
    public note!: any;
    public status!: any;
    public usermodifiedid!: any;
    public timemodified!: any;
    public timecreated!: any;
    public usercreatedid!: any;
    public dloaihinhkhuyenmaiid!: any;
    public tungay!: any;
    public denngay!: any;
    public ngungapdung!: any;
    public tilegiamgia!: any;
    public tilegiamgiatiengio!: any;
    public khuyenmaigiohat!: any;
    public tilegiamgiatong!: any;
    public tugio!: any;
    public dengio!: any;
    public tilegiamgiagiodau!: any;
}

DDOTKHUYENMAI.init({
    ID: {
        type: DataTypes.INTEGER,
        primaryKey: true, autoIncrement: true,
    },
    NAME: {
        type: DataTypes.STRING,
        
    },
    NOTE: {
        type: DataTypes.STRING,
        
    },
    STATUS: {
        type: DataTypes.BOOLEAN,
        
    },
    USERMODIFIEDID: {
        type: DataTypes.INTEGER,
        
    },
    TIMEMODIFIED: {
        type: DataTypes.DATE,
        
    },
    TIMECREATED: {
        type: DataTypes.DATE,
        
    },
    USERCREATEDID: {
        type: DataTypes.INTEGER,
        
    },
    DLOAIHINHKHUYENMAIID: {
        type: DataTypes.INTEGER,
        
    },
    TUNGAY: {
        type: DataTypes.DATE,
        
    },
    DENNGAY: {
        type: DataTypes.DATE,
        
    },
    NGUNGAPDUNG: {
        type: DataTypes.STRING,
        
    },
    TILEGIAMGIA: {
        type: DataTypes.DECIMAL(18,2),
        
    },
    TILEGIAMGIATIENGIO: {
        type: DataTypes.DECIMAL(18,2),
        
    },
    KHUYENMAIGIOHAT: {
        type: DataTypes.DATE,
        
    },
    TILEGIAMGIATONG: {
        type: DataTypes.DECIMAL(18,2),
        
    },
    TUGIO: {
        type: DataTypes.DATE,
        
    },
    DENGIO: {
        type: DataTypes.DATE,
        
    },
    TILEGIAMGIAGIODAU: {
        type: DataTypes.DECIMAL(18,2),
        
    },
}, {
    sequelize: db.sequelize,
    modelName: 'DDOTKHUYENMAI',
    tableName: 'DDOTKHUYENMAI',
    timestamps: false
});

export default DDOTKHUYENMAI;

File: frontend/Models/DDOTKHUYENMAI.cs
using System;

namespace QuanLyBar.Client.Models
{
    public class DDOTKHUYENMAI
    {
        public int? Id { get; set; }
        public string Name { get; set; }
        public string Note { get; set; }
        public bool? Status { get; set; }
        public int? UsermodifiedId { get; set; }
        public DateTime? Timemodified { get; set; }
        public DateTime? Timecreated { get; set; }
        public int? UsercreatedId { get; set; }
        public int? DloaihinhkhuyenmaiId { get; set; }
        public DateTime? Tungay { get; set; }
        public DateTime? Denngay { get; set; }
        public string Ngungapdung { get; set; }
        public decimal? Tilegiamgia { get; set; }
        public decimal? Tilegiamgiatiengio { get; set; }
        public DateTime? Khuyenmaigiohat { get; set; }
        public decimal? Tilegiamgiatong { get; set; }
        public DateTime? Tugio { get; set; }
        public DateTime? Dengio { get; set; }
        public decimal? Tilegiamgiagiodau { get; set; }
    }
}
